extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java

```diff
@@ -35,6 +39,30 @@ class FlywayCreatorTest {
      */
     private FlywayCreator creator;
 
+    @Inject
+    DataSource dataSource;
+
+    @Test
+    @DisplayName("Test Different user config")
+    void testDifferentUserConfig() throws SQLException {
+        FlywayDataSourceRuntimeConfig runtimeConfig = FlywayDataSourceRuntimeConfig.defaultConfig();
+        FlywayDataSourceBuildTimeConfig buildConfig = FlywayDataSourceBuildTimeConfig.defaultConfig();
+        runtimeConfig.username = runtimeConfig.username.of("sa");
+        runtimeConfig.password = runtimeConfig.password.of("sa");
+        runtimeConfig.jdbcUrl = runtimeConfig.jdbcUrl.of("jdbc:h2:~/test");
+        creator = new FlywayCreator(runtimeConfig, buildConfig);
+        Flyway flyway = creator.createFlyway(dataSource);
+        boolean error = false;
+        try {
+            flyway.validate();
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            error = true;
+        }
+
+        assertEquals(false, error, "Database connection with different user failed! ");
+    }
```

Comment:
I still don't think this test is right.
Ideally the test(s) should check if the Flyway configuration is using the expected JDBC URL, user and password when provided